Source/Editor/Viewport/MainEditorGizmoViewport.cs

```diff
@@ -576,15 +610,202 @@ protected override void OrientViewport(ref Quaternion orientation)
+                _rubberBandRect.Height = _viewMousePos.Y - _cachedStartingMousePosition.Y;
+
+                if (_lastRubberBandRect != _rubberBandRect)
+                {
+                    // Select rubberbanded rect actor nodes
+                    var adjustedRect = _rubberBandRect;
+                    _lastRubberBandRect = _rubberBandRect;
+                    if (adjustedRect.Width < 0 || adjustedRect.Height < 0)
+                    {
+                        // make sure we have a well-formed rectangle i.e. size is positive and X/Y is upper left corner
+                        var size = adjustedRect.Size;
+                        adjustedRect.X = Mathf.Min(adjustedRect.X, adjustedRect.X + adjustedRect.Width);
+                        adjustedRect.Y = Mathf.Min(adjustedRect.Y, adjustedRect.Y + adjustedRect.Height);
+                        size.X = Mathf.Abs(size.X);
+                        size.Y = Mathf.Abs(size.Y);
+                        adjustedRect.Size = size;
+                    }
+
+                    List<SceneGraphNode> hits = new List<SceneGraphNode>();
+                    var allActors = Level.GetActors<Actor>(true);
+                    foreach (var a in allActors)
+                    {
+                        if (a.HideFlags is HideFlags.DontSelect or HideFlags.FullyHidden || a is EmptyActor || a is Scene || !a.IsActive)
+                            continue;
+
+                        var actorBox = a.EditorBox;
+                        if (ViewFrustum.Contains(actorBox) == ContainmentType.Disjoint)
+                            continue;
+
+                        // Check if control and skip if canvas is 2D
+                        if (a is UIControl control)
+                        {
+                            UICanvas canvas = null;
+                            var controlParent = control.Parent;
+                            while (controlParent != null && controlParent is not Scene)
+                            {
+                                if (controlParent is UICanvas uiCanvas)
+                                {
+                                    canvas = uiCanvas;
+                                    break;
+                                }
+                                controlParent = controlParent.Parent;
+                            }
+
+                            if (canvas != null)
+                            {
+                                if (canvas.Is2D)
+                                    continue;
+                            }
+                        }
+                        else if (a is UICanvas uiCanvas)
+                        {
+                            if (uiCanvas.Is2D)
+                                continue;
+                        }
+
+                        var containsAllPoints = true;
+                        var fallBackToBox = false;
+                        if (a is StaticModel sm)
```

Comment:
A little documentation comment would help to understand why we do this mesh-specific code. Also, it would be nice to make it virtual call to `SceneGraphNode` so other models could implement or customize it in future.

Reply:
Done